elastica/rigidbody/data_structures.py

```diff
@@ -51,83 +49,83 @@ def __init__(self) -> None:
-    (references) from it, and outputs State classes that are used in the time-stepping
-    algorithm. This means that modifying the state modifies the internal data!
-
-    Parameters
-    ----------
-    stepper_type : str (likely to change in future), representing stepper type
-    Allowed parameters are ['explicit', 'symplectic']
-    n_elems : int, number of rod elements
-    vector_states : np.ndarray of shape (dim, *) with the following structure
-        `vector_states` = [`position`,`velocity`,`omega`,`acceleration`,`angular acceleration`]
-        `n_nodes = n_elems =  1`
-        `position = 0 -> n_nodes , size = n_nodes`
-        `velocity = n_nodes -> 2 * n_nodes, size = n_nodes`
-        `omega = 2 * n_nodes -> 2 * n_nodes + nelem, size = nelem`
-        `acceleration = 2 * n_nodes + nelem -> 3 * n_nodes + nelem, size = n_nodes`
-        `angular acceleration = 3 * n_nodes + nelem -> 3 * n_nodes + 2 * nelem, size = n_elems`
-    matrix_states : np.ndarray of shape (dim, dim, n_elems) containing the directors
-
-    Returns
-    -------
-    output : tuple of len 8 containing
-    (state, derivative_state, position, directors, velocity, omega, acceleration, alpha)
-    derivative_state carries rate information
-
-    """
-    n_nodes = n_elems
-    position = np.ndarray.view(vector_states[..., :n_nodes])
-    directors = np.ndarray.view(matrix_states)
-    v_w_dvdt_dwdt = np.ndarray.view(vector_states[..., n_nodes:])
-    output = ()
-    # TODO:
-    # 11/01/2020: Extend Rigid body data structures for Explicit steppers
-    # 12/20/2021: Future work! If explicit stepper is gonna be dropped, remove.
-    # if stepper_type == "explicit":
-    #     v_w_states = np.ndarray.view(vector_states[..., n_nodes : n_nodes + 1])
-    #     output += (
-    #         _State(n_elems, position, directors, v_w_states),
-    #         _DerivativeState(n_elems, v_w_dvdt_dwdt),
-    #     )
-    # elif stepper_type == "symplectic":
-    #     output += (
-    #         _KinematicState(n_elems, position, directors),
-    #         _DynamicState(n_elems, v_w_dvdt_dwdt),
-    #     )
-    # else:
-    #     return
-    output += (
-        _KinematicState(n_elems, position, directors),
-        _DynamicState(n_elems, v_w_dvdt_dwdt),
-    )
-
-    n_velocity_end = n_nodes + n_nodes
-    velocity = np.ndarray.view(vector_states[..., n_nodes:n_velocity_end])
-
-    n_omega_end = n_velocity_end + n_elems
-    omega = np.ndarray.view(vector_states[..., n_velocity_end:n_omega_end])
-
-    n_acceleration_end = n_omega_end + n_nodes
-    acceleration = np.ndarray.view(vector_states[..., n_omega_end:n_acceleration_end])
-
-    n_alpha_end = n_acceleration_end + n_elems
-    alpha = np.ndarray.view(vector_states[..., n_acceleration_end:n_alpha_end])
-
-    return output + (position, directors, velocity, omega, acceleration, alpha)
+# def _bootstrap_from_data(stepper_type: str, n_elems: int, vector_states, matrix_states):
```

Comment:
Just remove all comments below. I feel like `data_structures` needs to move to timestepper, and both `rigid_body` and `rod` can import from same file. I can refactor that later with rod module.

Reply:
[d7da645](https://github.com/GazzolaLab/PyElastica/pull/395/commits/d7da64525e2ceacd6c3c246407bbf4ec42e708e8)

Reply:
[d7da645](https://github.com/GazzolaLab/PyElastica/pull/395/commits/d7da64525e2ceacd6c3c246407bbf4ec42e708e8)